pkg/deploy/generator/resources.go

```diff
@@ -22,7 +22,7 @@ import (
 )
 
 var apiVersions = map[string]string{
-	"authorization": "2018-09-01-preview",
+	"authorization": "2018-07-01",
```

Comment:
Can't do this because you're still importing 2018-09-01-preview above.  The Go SDK is limited in the versions that it provides.  I think you need to roll the auth back to 2015-07-01.  Below, that'll imply diffs like:
```diff
                        Resource: &mgmtauthorization.RoleAssignment{
                                Name: to.StringPtr("[guid(resourceGroup().id, 'RP / Reader')]"),
                                Type: to.StringPtr("Microsoft.Authorization/roleAssignments"),
-                               RoleAssignmentPropertiesWithScope: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
+                               Properties: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
                                        Scope:            to.StringPtr("[resourceGroup().id]"),
                                        RoleDefinitionID: to.StringPtr("[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48e>
                                        PrincipalID:      to.StringPtr("[parameters('rpServicePrincipalId')]"),
-                                       PrincipalType:    mgmtauthorization.ServicePrincipal,
                                },
                        },
```

Reply:
There is another problem here, as this will not work eitherway. WIll do 2 PR's.